components/properties/src/maps.rs

```diff
@@ -216,3 +216,27 @@ where
 {
     get_cp_map(provider, key::SENTENCE_BREAK_V1)
 }
+
+/// Return a [`CodePointTrie`] for the Canonical_Combining_Class Unicode enumerated
```

Comment:
Praise: this looks correct to classify the Canonical_Combining_Class as an enumerated property.

Even though UAX 44 says that Canonical_Combining_Class ["could be considered either an enumerated property or a numeric property: the principal use of the property is in terms of the numeric values,"](https://unicode.org/reports/tr44/#Canonical_Combining_Class), ICU treats it as an enumerated property in [uchar.h](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/uchar_8h_source.html) and then uses the integer values (aka enum discriminants) for algorithmic purposes.

Reply:
ICU has UCHAR_CANONICAL_COMBINING_CLASS=0x1002 in among the "enumerated/integer Unicode properties", which means that ICU by-code point functions return int values rather than boolean or string or something else. However, for all intents and purposes, it's really a numeric property. Pretty much any runtime use will want to see integers coming out of the trie, for use in canonical reordering, or for checking if a string is already in that order. Having to unpack an enum to get its value would be awkward. (I assume that it would be more than a name for an integer, which is what a C enum constant is.)

Note that the integer values that you get do not form a contiguous range. (Unlike the ordinals of a manually maintained enum.)

Note also that the Unicode maintainers for a while were using more and more values but didn't bother giving them descriptive names. As a formality, just to list names in PropertyValueAliases.txt, they ended up with not-really-names like "CCC36" for numeric value 36.

Yes, there are short and long names for those numeric values that are in actual use, and ICU supports mapping between these name strings and the numeric values, but we don't even have an enum with named constants, and no one has ever asked for one.

So I propose that this trie simply map code points to some integer type. `u8` is big enough. (I don't know if Rust developers always choose the smallest type. Java & C++ developers tend to prefer int or int32_t.)

Here in the docs, I would simply drop the "enumerated" and just say "... for the Canonical_Combining_Class Unicode property."

Reply:
I think the Rust preference is the smallest type and having it newtyped, since the other properties are newtyped, too. See my other comment about `Ord` and `PartialOrd` about why the newtyping doesn't interfere with ergonomics in practice.

Reply:
ok

Reply:
+1

- Smaller types are better because they impact data size
- Newtypes are a zero-cost abstraction in Rust and give better type-safety and ergonomics